source/MaterialXGenMdl/MdlSyntax.cpp

```diff
@@ -170,8 +170,8 @@ const StringVec MdlSyntax::VECTOR2_MEMBERS = { ".x", ".y" };
 const StringVec MdlSyntax::VECTOR3_MEMBERS = { ".x", ".y", ".z" };
 const StringVec MdlSyntax::VECTOR4_MEMBERS = { ".x", ".y", ".z", ".w" };
 const StringVec MdlSyntax::COLOR2_MEMBERS = { ".x", ".y" };
-const StringVec MdlSyntax::COLOR3_MEMBERS = { "[0]", "[1]", "[2]" };
-const StringVec MdlSyntax::COLOR4_MEMBERS = { ".rgb[0]", ".rgb[1]", ".rgb[2]", ".a" };
+const StringVec MdlSyntax::COLOR3_MEMBERS = { ".x", ".y", ".z" };
```

Comment:
This is still not correct. We'll need a custom syntax handler for this to convert from color3 to float3 then use these accessors. (Not done yet).

Reply:
I have added handling of this on my end, with custom syntax handling for color3/color4, so that swizzling and using the channels attribute works again. So should be ok when we merge.